src/core/gba/gbaElf.cpp

```diff
@@ -2044,14 +2594,35 @@ uint8_t* elfParseBlock(uint8_t* data, ELFAbbrev* abbrev, CompileUnit* unit,
                 data = elfSkipData(data, abbrev, unit->abbrevs);
             } break;
             }
+            
+            if (abbrev->hasChildren) {
+                nesting++;
+                depth++;
+            }
+        }
+        
+        if (depth >= MAX_BLOCK_NESTING) {
+            fprintf(stderr, "Maximum block nesting depth exceeded\n");
         }
     }
     return data;
 }
 
 void elfGetFunctionAttributes(CompileUnit* unit, uint32_t offset, Function* func)
 {
+    // Validate offset is within compile unit
+    if (offset > unit->length) {
+        fprintf(stderr, "Error: Function attribute offset %u exceeds unit length %u\n", offset, unit->length);
+        return;
```

Comment:
For these kind of functions, I wonder if we should have a global `bool g_elfHasError`, and set that to true so we don't continue at the call site.